picard/file.py

```diff
@@ -351,7 +351,13 @@ def _script_to_filename(self, naming_format, file_metadata, settings=None):
         else:
             metadata.copy(self.orig_metadata)
             metadata.update(file_metadata)
-        return script_to_filename(naming_format, metadata, file=self, settings=settings)
+        (filename, new_metadata) = script_to_filename(naming_format, metadata,
+                                                      file=self, settings=settings)
+        # NOTE: the script_to_filename strips the extension away
+        ext = new_metadata.get('~extension', file_extension)
+        if not ext.startswith('.'):
```

Comment:
I am a bit bothered by this as well, but I think we need to check that. For one there is a difference between the ext generated basically with `_fixed_splitext` and  the value stored in `~extension` in metadata.

We could of course change `_fixed_splitext` to return the extension without the dot. But that's basically calling `os.path.splitext` (which returns the dot), and we would just remove the dot to add it again basically directly afterwards. That's a bit silly.

The one in metadata is by definition without the dot because we set it that way. We could of course change it and set `~extension` always with a dot, but this would for one break existing scripts and also I think it is nicer for the user not to bother with the dot in scripts. In addition since we read this from script now the user can end up changing it.

What we maybe could do instead of 

```
if not ext.startswith('.'):
    ext = '.' + ext
return filename + ext
```

is something like:

```
return filename + '.' + ext.lstrip('.')
```

Not sure if that's much better, though.

Reply:
Ok, messy ;)

I prefer:

```python
return filename + '.' + ext.lstrip('.')
```

Reply:
`return filename + '.' + ext.lstrip('.')` is the best way imho.

Reply:
Ok, your get this in 162d081af5d49102a435ea4e19dd84a7ce1651bd :D